Initial

otchappi opened the pull request.

kilian-goetz committed:
Add MUI library

kilian-goetz committed:
Delete default skeleton

kilian-goetz committed:
Create navbar components

kilian-goetz committed:
Install react router dom package for routes

kilian-goetz committed:
Create skeleton of the project

Create an empty home page an a route just to define the structure of the frontend project

kilian-goetz committed:
Create drawer, routes and components

kilian-goetz committed:
Remove connection button

Connection button is useless because we need to be connected to access website.

kilian-goetz committed:
Create login page

Valentin271 committed:
chore: added ci

kilian-goetz committed:
Replace window location method by react router dom redirect method

otchappi committed:
initialize

otchappi committed:
Lint fix

otchappi committed:
Fix lint

otchappi committed:
fix build

otchappi committed:
Mission type : delete, pagination

otchappi committed:
Block nurse

kilian-goetz committed:
Heavy refactor, still in progress

Delete useless depedencies, delete useless css properties or html/react tags, simplify lot of codes, etc.

kilian-goetz committed:
Delete useless depedency

I forgot one depedency

kilian-goetz committed:
Delete useless code

kilian-goetz committed:
Delete useless code

kilian-goetz committed:
Delete useless file

kilian-goetz committed:
Refactor structure of some files

kilian-goetz committed:
Refactor interface name

kilian-goetz committed:
Delete useless lines

kilian-goetz committed:
Create all fetch methods for managers

kilian-goetz committed:
Create all fetch methods for centers

kilian-goetz committed:
Create all fetch methods for missions

kilian-goetz committed:
Create all fetch methods for nurses

kilian-goetz committed:
Fix missing issue

kilian-goetz committed:
Create all fetch methods for patients

kilian-goetz committed:
Create all fetch methods for skills

kilian-goetz committed:
Create all fetch methods for visits

kilian-goetz committed:
Create all fetch methods for auth

kilian-goetz committed:
Delete useless file

formatterUtils file was deleted because it's possible with mui to define a minimum for textfield type number, so a simple parseInt is enough

kilian-goetz committed:
Create not found page and improve redirection

kilian-goetz committed:
Refactor some files name

kilian-goetz committed:
Refactor som files name

Valentin271 commented on src/components/navbarComponent/navbarComponent.tsx:
```diff
@@ -0,0 +1,236 @@
+import { blue } from '@mui/material/colors';
+import PageTitleComponent from '../pageTitleComponent/pageTitleComponent.tsx';
+import {useAuth} from "../../context/auth/authContext.ts";
+import {ReactElement} from "react";
```
Mui components should be grouped

Valentin271 commented on src/context/auth/authContext.ts:
```diff
@@ -0,0 +1,25 @@
+
+export interface IFormValues{
+    mail: string,
+    password: string,
```
Non! Pas le password dans le context

Valentin271 commented on src/utils/request.ts:
```diff
@@ -0,0 +1,98 @@
+    }).then((res) => {
+        switch (res.status) {
+            case 401:
+                //window.location.assign("/login");
```
A remettre

kilian-goetz committed:
Change redirect logic and not found page

kilian-goetz committed:
Refactor pages and path

kilian-goetz committed:
Refactor missions table and connect api

kilian-goetz committed:
Regroup imports in navbarComponent

kilian-goetz committed:
Type getAllMission fetch method

kilian-goetz committed:
Fix lint error

kilian-goetz committed:
Refactor date in mission table

kilian-goetz committed:
Fix mission type table issues

Mission type table did not display the required number of people and the duration. Additionally, I changed the colors of the icons to make it less gloomy.

kilian-goetz committed:
Many new features

Add actions to mission table (edit, delete and create), only deletion is available. Centered tabs in profile page.

kilian-goetz committed:
Rename pages files to have consistent names

kilian-goetz committed:
Refactor mission type table duration to be in XXhXXm format

kilian-goetz committed:
Change favicon and website name

kilian-goetz committed:
Turn fullcalendar into french

kilian-goetz committed:
Refactor missionPage

I reduced the number of useless lines by map component

kilian-goetz committed:
Divide navbar into three files to be more consistent

I divided navabr component intro three files, one for drawer, one for app bar and last one to regroup both. I do so because navabr was import unreadable

kilian-goetz committed:
I added the initials of the current user to the navigation bar's avatar

kilian-goetz committed:
Refactor AuthContext

AuthContext contain no longer password and mail. Now AuthContext contains isLogin to help redirection and UserInfo

Valentin271 committed:
chore: added prettier

Valentin271 committed:
chore: fmt
